File: DemoCode/Compilation/OffScreenRendering/ViewController.m
//
//  ViewController.m
//  OffScreenRendering
//
//  Created by super on 2019/1/9.
//  Copyright © 2019 Jeremy. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIButton *button = ({
        UIButton * btn = [UIButton buttonWithType:UIButtonTypeCustom];
        btn.frame = CGRectMake(100, 10, 100, 100);
//        btn.backgroundColor = [UIColor blackColor];
        btn.layer.cornerRadius = 50;
        [btn setImage:[UIImage imageNamed:@"sao.jpg"] forState:UIControlStateNormal];
//        btn.layer.masksToBounds = YES;
        [self.view addSubview:btn];
        btn.clipsToBounds = YES;
        btn;
    });
    
    UIButton *button1 = ({
        UIButton * btn = [UIButton buttonWithType:UIButtonTypeCustom];
        btn.frame = CGRectMake(100, 150, 100, 100);
        btn.backgroundColor = [UIColor blackColor];
        btn.layer.cornerRadius = 50;
        btn.layer.masksToBounds = YES;
        [self.view addSubview:btn];
        btn.clipsToBounds = YES;
        btn;
    });
    
    UIImageView * imageView = ({
       UIImageView * imageView = [[UIImageView alloc] init];
        imageView.frame = CGRectMake(100, 260, 100, 100);
        imageView.backgroundColor = [UIColor blackColor];
        imageView.layer.cornerRadius = 50;
        imageView.layer.masksToBounds = YES;
        imageView.image = [UIImage imageNamed:@"sao.jpg"];
        [self.view addSubview:imageView];
        imageView;
    });
    
    UIImageView * imageView1 = ({
        UIImageView * imageView = [[UIImageView alloc] init];
        imageView.frame = CGRectMake(100, 400, 100, 100);
//        imageView.backgroundColor = [UIColor blackColor];
        imageView.layer.cornerRadius = 50;
//        imageView.layer.masksToBounds = YES;
        [self.view addSubview:imageView];
        imageView.image = [UIImage imageNamed:@"sao.jpg"];
        imageView;
    });
    
    UIImageView * imageView2 = ({
        UIImageView * imageView = [[UIImageView alloc] init];
        imageView.frame = CGRectMake(100, 550, 100, 100);
//        imageView.backgroundColor = [UIColor blackColor];
        imageView.layer.cornerRadius = 50;
        imageView.layer.masksToBounds = YES;
        imageView.image = [UIImage imageNamed:@"sao.jpg"];
        [self.view addSubview:imageView];
        imageView;
    });
}


@end

File: DemoCode/Compilation/Navgation/ViewController.swift
//
//  ViewController.swift
//  Navgation
//
//  Created by Jeremy on 2019/2/15.
//  Copyright © 2019 Jeremy. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UISearchControllerDelegate {
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "设置"
        let sc = UISearchController.init(searchResultsController: nil)
        navigationItem.searchController = sc
        navigationItem.hidesSearchBarWhenScrolling = true
        let tableView = UIScrollView.init(frame: view.bounds)
        tableView.contentSize = CGSize.init(width: tableView.frame.size.width, height: 1000)
        self.view.addSubview(tableView)
        
        sc.delegate = self as UISearchControllerDelegate
        let scb = sc.searchBar
        scb.tintColor = UIColor.white
        scb.barTintColor = UIColor.white
        
        if let textfield = scb.value(forKey: "searchField") as? UITextField {
            textfield.textColor = UIColor.blue
            if let backgroundview = textfield.subviews.first {
                
                // Background color
                backgroundview.backgroundColor = UIColor.white
                
                // Rounded corner
                backgroundview.layer.cornerRadius = 10;
                backgroundview.clipsToBounds = true;
                
            }
        }
        
        if let navigationbar = self.navigationController?.navigationBar {
            navigationbar.barTintColor = UIColor.blue
        }
    }
}

